x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlResponseListener.java

```diff
@@ -229,4 +232,16 @@ private void checkDelimiter() {
             throw new IllegalArgumentException(message);
         }
     }
+
+    /**
+     * Log all partial request failures to the {@code rest.suppressed} logger
+     * so an operator can categorize them after the fact.
+     */
+    static void logPartialFailures(String rawPath, Map<String, String> params, EsqlExecutionInfo exeuctionInfo) {
+        for (EsqlExecutionInfo.Cluster cluster : exeuctionInfo.getClusters().values()) {
+            for (ShardSearchFailure failure : cluster.getFailures()) {
+                RestResponse.logSuppressedError(org.apache.logging.log4j.Level.WARN, rawPath, params, RestStatus.OK, failure);
```

Comment:
May it be useful to also log which cluster the error came from?

Reply:
Checking if this does that already.